### PR TITLE
chore: combining linux and win32 execution engine param structures.

### PR DESCRIPTION
**Pull request by jebrando**

### CODE REVIEW

File: interfaces/inc/c_pal/execution_engine.h [hunk 1]
```diff
@@ -3,12 +3,27 @@
...
+    uint32_t min_thread_count;
+    uint32_t max_thread_count;
+} EXECUTION_ENGINE_PARAMETERS;
+
+#define DEFAULT_MIN_THREAD_COUNT 4
+#define DEFAULT_MAX_THREAD_COUNT 0 // no max thread count
+
 typedef struct EXECUTION_ENGINE_TAG* EXECUTION_ENGINE_HANDLE;
 
 MOCKABLE_FUNCTION(, EXECUTION_ENGINE_HANDLE, execution_engine_create, void*, execution_engine_parameters);
```
Comment (anporumb) [inline interfaces/inc/c_pal/execution_engine.h:29]: >void [](http://example.com/codeflow?start=70&length=4)

isn't this then const `EXECUTION_ENGINE_PARAMETERS`*?

**Comment by anporumb** [comment]
> MOCKABLE_FUNCTION(, void, execution_engine_dec_ref, EXECUTION_ENGINE_HANDLE, execution_engine);

also THANDLE it :)

---
Refers to: interfaces/inc/c_pal/execution_engine.h:30 in 867e957. [](commit_id = 867e95796a9e0522fb51b4df8415655423c8d24c, deletion_comment = False)

**Review by anporumb — APPROVED** [review]
:shipit:

File: interfaces/inc/c_pal/execution_engine.h (hunk 1 above)
Comment (jebrando) [reply interfaces/inc/c_pal/execution_engine.h:29]: I believe it can be.

**Comment by jebrando** [comment]
> MOCKABLE_FUNCTION(, void, execution_engine_dec_ref, EXECUTION_ENGINE_HANDLE, execution_engine);

The pointer is not saved, so there's no reason for a THANDLE here.

---
In reply to: [1701824978](https://github.com/Azure/c-pal/pull/265#issuecomment-1701824978) [](http://example.com/codeflow?ancestors=1701824978,1701824978)

---
Refers to: interfaces/inc/c_pal/execution_engine.h:30 in 867e957. [](commit_id = 867e95796a9e0522fb51b4df8415655423c8d24c, deletion_comment = False)

**Pull request merged**